src/executables/master_server_monitor.cpp

```diff
@@ -52,18 +51,42 @@ server_monitor::~server_monitor ()
   if (m_monitoring_thread->joinable())
     {
       m_monitoring_thread->join();
-      fprintf (stdout, "server_monitor_thread is terminated. \n");
+      fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_monitor_thread is terminated. \n");
     }
 
   assert (m_server_entry_list->size () == 0);
-  fprintf (stdout, "server_entry_list is deleted. \n");
+  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_entry_list is deleted. \n");
   fflush (stdout);
 }
 
+void
+server_monitor::make_and_insert_server_entry (int pid, const char *exec_path, char *args,
+    CSS_CONN_ENTRY *conn)
+{
+  m_server_entry_list->emplace_back (pid, exec_path, args, conn);
+  fprintf (stdout,
+	   "[SERVER_REVIVE_DEBUG] : server has been registered into master_Server_monitor : pid : %d, exec_path : %s, args : %s\n",
+	   pid,
+	   exec_path, args);
+}
+
+void
+server_monitor::remove_server_entry_by_conn (CSS_CONN_ENTRY *conn)
+{
+  const auto result = std::remove_if (m_server_entry_list->begin(), m_server_entry_list->end(),
+				      [conn] (auto e) -> bool {return e.m_conn == conn;});
+#if defined(DEBUG)
+  assert (result != m_server_entry_list->end ());
+#endif
```

Comment:
```suggestion
  assert (result != m_server_entry_list->end ());
```

assert () 는 debug 모드에서만 호출됩니다.

Reply:
`defined(DEBUG)` 조건 제거하였습니다.